test/jdk/java/foreign/channels/TestAsyncSocketChannels.java

```diff
@@ -279,15 +280,40 @@ public void completed(Long result, Void att) {
             // in turn unlock the scope and allow it to be closed.
             readNBytes(asc2, bytesWritten.get());
             assertTrue(scope.isAlive());
-            out.println("outstanding writes: " + outstandingWriteOps.get());
-            while (outstandingWriteOps.get() > 0 )  {
-                out.println("spinning");
-                Thread.onSpinWait();
-            }
+            awaitOutstandingWrites(outstandingWriteOps);
```

Comment:
Strictly speaking it wasn't necessary to touch this area of code to resolve the failures that we're seeing - I decided to refactor this into a separate method to improve readability of the test logic. Here again, awaiting for completion is not strictly necessary, just good practice to ensure that all threads and operations associated with the test scenario complete before the next scenario. I'll see if this can be simplified as you suggest.